server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -285,9 +278,10 @@ private MergeResult partialReduce(
             }
             // we have to merge here in the same way we collect on a shard
             newTopDocs = topDocsList == null ? null : mergeTopDocs(topDocsList, topNSize, 0);
-            newAggs = aggsList == null
-                ? null
-                : InternalAggregations.topLevelReduceDelayable(aggsList, aggReduceContextBuilder.forPartialReduction());
+            newAggs = hasAggs
```

Comment:
I wonder if there is an exception in the middle of processing aggs, we might be releasing toConsume twice, is that ok?

Reply:
Yea I'd say so, releasing aggs is idempotent since we null out the aggs field when we do :)

Reply:
I see that you are nullifying them now as they are consumed.